### PR TITLE
Add a content author guide #234

### LINKED ISSUE #234: Add a content author guide
A guide to explain the syntax for commonly used elements such as embedded links, tooltips, useful style classes etc.

### PR DESCRIPTION
**Pull request by acjh**
Fixes #234 
[Preview](https://github.com/acjh/website/blob/234-add-content-author-guide/doc/ContentAuthorGuide.md)

### CODE REVIEW

**Comment by damithc** [comment]
Looks good. Other things you can include:
* additional-info class
* `<code>` tag
* How to use syntax highlighting for code samples

**Comment by acjh** [comment]
@damithc Ready for review.

File: doc/ContentAuthorGuide.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+Adding an embedded link:
+
+1. Surround with `<span onclick="getContentUsingAjax('param-1', 'param-2', 'param-3');" class="embedded-link">`**Link Text**`</span>`
+where:
+    * param-1: filename (without .html)
+    * param-2: #selector-for-element-in-current-page-to-load-into (with #)
+    * param-3: section id for element in param-1 to load-into param-2 (optional)
+
+e.g. `<span onclick="getContentUsingAjax('handbook-teams', '#embedded-week1-teams');" class="embedded-link">`**Teams**`</span>`
```
Comment (damithc) [inline doc/ContentAuthorGuide.md]: Embedded links has another part: the div that marks where to load the content. Usually it appears slightly below the link.

Comment (acjh) [inline doc/ContentAuthorGuide.md]: Will add :)

Comment (damithc) [inline doc/ContentAuthorGuide.md]: Also, a link from README to this doc can be added.

File: doc/ContentAuthorGuide.md [hunk 2]
```diff
@@ -19,13 +19,16 @@ Embedded links are used to load supplementary sections, triggered by a click.
...
 Adding an embedded link:
 
-1. Surround with `<span onclick="getContentUsingAjax('param-1', 'param-2', 'param-3');" class="embedded-link">`**Link Text**`</span>`
-where:
-    * param-1: filename (without .html)
-    * param-2: #selector-for-element-in-current-page-to-load-into (with #)
-    * param-3: section id for element in param-1 to load-into param-2 (optional)
-
-e.g. `<span onclick="getContentUsingAjax('handbook-teams', '#embedded-week1-teams');" class="embedded-link">`**Teams**`</span>`
+1. Enclose link text in `<span onclick="getContentUsingAjax('param-1', 'param-2', 'param-3');" class="embedded-link">` `</span>`,
```
Comment (damithc) [inline doc/ContentAuthorGuide.md]: perhaps `param-name` etc.

File: doc/ContentAuthorGuide.md [hunk 3]
```diff
@@ -19,13 +19,16 @@ Embedded links are used to load supplementary sections, triggered by a click.
...
-    * param-1: filename (without .html)
-    * param-2: #selector-for-element-in-current-page-to-load-into (with #)
-    * param-3: section id for element in param-1 to load-into param-2 (optional)
-
-e.g. `<span onclick="getContentUsingAjax('handbook-teams', '#embedded-week1-teams');" class="embedded-link">`**Teams**`</span>`
+1. Enclose link text in `<span onclick="getContentUsingAjax('param-1', 'param-2', 'param-3');" class="embedded-link">` `</span>`,  
+   e.g. `<span onclick="getContentUsingAjax('handbook-teams', '#embedded-week1-teams');" class="embedded-link">`**Teams**`</span>`
+    * param-1: name of the file to load from (without .html)
+    * param-2: `#selector` for element in this page to load into
+    * param-3: section id for element in file to load from (optional)
```
Comment (damithc) [inline doc/ContentAuthorGuide.md]: IIRC, there is an overloaded version of this method that takes another parameter: title to show in the embedded content box

File: doc/ContentAuthorGuide.md (hunk 2 above)
Comment (acjh) [inline doc/ContentAuthorGuide.md]: `param-name`, `param-selector`, `param-id`?

File: doc/ContentAuthorGuide.md (hunk 3 above)
Comment (acjh) [inline doc/ContentAuthorGuide.md]: That's `pull-content(_,_,_,_)`. I can include that.

File: doc/ContentAuthorGuide.md (hunk 2 above)
Comment (damithc) [inline doc/ContentAuthorGuide.md]: Yes, something like that

File: doc/ContentAuthorGuide.md (hunk 3 above)
Comment (damithc) [inline doc/ContentAuthorGuide.md]: Yes, that one.

**Comment by acjh** [comment]
Changes made, ready for review.

**Pull request merged**